src/av2/datasets/motion_forecasting/eval/metrics.py

```diff
@@ -56,3 +55,85 @@ def compute_is_missed_prediction(
+        forecast_probabilities: (K,) probabilities associated with each prediction.
+        normalize: Normalizes `forecast_probabilities` to sum to 1 when set to True.
+
+    Raises:
+        ValueError: If the number of forecasted trajectories and probabilities don't match.
+        ValueError: If normalize=False and `forecast_probabilities` contains values outside of the range [0, 1].
+
+    Returns:
+        (K,) Probability-weighted average displacement error for each predicted trajectory.
+    """
+    # Validate that # of forecast probabilities matches forecasted trajectories
+    if len(forecasted_trajectories) != len(forecast_probabilities):
+        raise ValueError()
+
+    # If enabled, normalize forecast probabilities to sum to 1
+    if normalize:
+        forecast_probabilities = forecast_probabilities / np.sum(forecast_probabilities)
+
+    # Validate that all forecast probabilities are in the range [0, 1]
+    if np.logical_or(forecast_probabilities < 0.0, forecast_probabilities > 1.0).any():
+        raise ValueError("At least one forecast probability falls outside the range [0, 1].")
+
+    # Compute ADE with Brier score component
+    ade_vector = compute_ade(forecasted_trajectories, gt_trajectory)
+    brier_score = np.square((1 - forecast_probabilities))
+    brier_ade: NDArrayFloat = ade_vector + brier_score
+    return brier_ade
+
+
+def compute_brier_fde(
+    forecasted_trajectories: NDArrayNumber,
+    gt_trajectory: NDArrayNumber,
+    forecast_probabilities: NDArrayNumber,
+    normalize: bool = False,
+) -> NDArrayFloat:
+    """Compute a probability-weighted (using Brier score) FDE for K predicted trajectories (for the same actor).
+
+    Args:
+        forecasted_trajectories: (K, N, 2) predicted trajectories, each N timestamps in length.
+        gt_trajectory: (N, 2) ground truth trajectory.
+        forecast_probabilities: (K,) probabilities associated with each prediction.
+        normalize: Normalizes `forecast_probabilities` to sum to 1 when set to True.
+
+    Raises:
+        ValueError: If the number of forecasted trajectories and probabilities don't match.
+        ValueError: If normalize=False and `forecast_probabilities` contains values outside of the range [0, 1].
+
+    Returns:
+        (K,) Probability-weighted final displacement error for each predicted trajectory.
+    """
+    # Validate that # of forecast probabilities matches forecasted trajectories
+    if len(forecasted_trajectories) != len(forecast_probabilities):
+        raise ValueError()
+
+    # If enabled, normalize forecast probabilities to sum to 1
+    if normalize:
+        forecast_probabilities = forecast_probabilities / np.sum(forecast_probabilities)
+
+    # Validate that all forecast probabilities are in the range [0, 1]
+    if np.logical_or(forecast_probabilities < 0.0, forecast_probabilities > 1.0).any():
+        raise ValueError("At least one forecast probability falls outside the range [0, 1].")
+
```

Comment:
The two functions differ by just 1 line. Maybe move most of the stuff to a common function.